Configure Chai and testing database integration

pedro-m-dietrich opened the pull request:
When running `npm test`, the `test.env` file will be used to fetch the environment variables, instead of the `.env` (used for development).

A test case for checking the connection with the DB has been added, by creating an user directly in the database using **Prisma**.
This will be replaced by proper query/mutation tests later.

pedro-m-dietrich committed:
Use Chai package for assertion

To improve testing, the package `chai` will be used for assertions on
tests.

pedro-m-dietrich committed:
Configure test database

When running the tests, the correct .env (test.env) file must be used.

A test case for creating a user with Prisma has been added to verify the
database connection.

alanraso commented on src/server.ts:
```diff
@@ -4,7 +4,11 @@ import { startStandaloneServer } from '@apollo/server/standalone';
+
+export const startDatabase = (): void => {
+  prisma = new PrismaClient();
+};
```
O nome `startDatabase` parce pra mim que ele já vai conectar com o banco. Talvez algo mais específico como `initializeDatabaseInstance`

matheusflds commented on .gitignore:
```diff
@@ -6,4 +6,4 @@ dist/
 db/
 
-.env
+*.env
```
acho que pode manter só o .env no gitignore
via de regra commitamos o test.env mesmo (até pq é pra ter só valores de teste ou fakes la) para poder reaproveitar nas etapas de CI ou para outra pessoa clonar o repo e testar fácil, faz sentido?

pedro-m-dietrich replied:
Faz sentido sim.
Já apliquei a mudança.

pedro-m-dietrich committed:
Refactor database instance initialization

A better name is used for initializing the Prisma instance.

The `test.env` file will be commited as it does not contain sensitive
information and can help the testing CI stages.
